app/classifier/drawing-tools/column-rectangle.cjsx

```diff
@@ -40,20 +42,27 @@ module.exports = React.createClass
 
   initCoords: null
 
+  dragEnd: (target) ->
+    @context.geordi?.logEvent type: 'drag-column'
+    deleteIfOutOfBounds target
+
+  logResize: ->
+    @context.geordi?.logEvent type: 'resize-column'
+
   render: ->
     {x, width} = @props.mark
 
     <DrawingToolRoot tool={this}>
-      <Draggable onDrag={@handleMainDrag} onEnd={deleteIfOutOfBounds.bind null, this} disabled={@props.disabled}>
+      <Draggable onDrag={@handleMainDrag} onEnd={@dragEnd.bind this, this} disabled={@props.disabled}>
```

Comment:
React [autobinds for the mouse event](http://facebook.github.io/react/blog/2013/07/02/react-v0-4-autobind-by-default.html) so you can call the event handler and get what you'd expect back in a normal DOM event. However, if you want to do something with the React class instead, you have to bind it: `handler.bind(null, this)` and `this` will be the React class rather than the mouse event. Using `handler.bind(this)` will throw a warning in the console because the React library thinks you're trying to do the same thing as the autobinding. I'm not sure why this was changed to `@dragEnd.bind this, this`, though, @amyrebecca any clarification? FWIW, it seems like they accomplish the same thing.